### PR TITLE
91: Add support for mandatory flag. Make sure it works for RpcClient

### PR DESCRIPTION
**Pull request by janssk1**
Change-Id: I230ab88df99b665c47329caa6e8c89ba1af016cf

### CODE REVIEW

File: src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java [hunk 1]
```diff
@@ -16,14 +16,7 @@
...
 
-import com.rabbitmq.client.AMQP;
-import com.rabbitmq.client.BuiltinExchangeType;
-import com.rabbitmq.client.Channel;
-import com.rabbitmq.client.Connection;
-import com.rabbitmq.client.DefaultConsumer;
-import com.rabbitmq.client.Envelope;
-import com.rabbitmq.client.GetResponse;
-import com.rabbitmq.client.ShutdownSignalException;
+import com.rabbitmq.client.*;
```
Comment (ledoyen) [inline src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.rabbitmq.client'](https://app.codacy.com/manual/ledoyen/rabbitmq-mock/pullRequest?prid=4930446)

File: src/main/java/com/github/fridujo/rabbitmq/mock/MockChannel.java [hunk 2]
```diff
@@ -1,37 +1,17 @@
...
-import java.util.Map;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
 
+import com.rabbitmq.client.*;
```
Comment (ledoyen) [inline src/main/java/com/github/fridujo/rabbitmq/mock/MockChannel.java]: Codacy Issue found: [Avoid unused imports such as 'com.rabbitmq.client'](https://app.codacy.com/manual/ledoyen/rabbitmq-mock/pullRequest?prid=4930446)

File: src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java [hunk 3]
```diff
@@ -713,6 +724,126 @@ public void handleCancelOk(String consumerTag) {
...
+    @Test
+    void mandatory_publish_with_multiple_listeners() throws IOException, TimeoutException {
+        try (Connection conn = new MockConnectionFactory().newConnection()) {
+            try (Channel channel = conn.createChannel()) {
+                TrackingCallback firstCallbackThatIsRegistedTwice = new TrackingCallback();
+                TrackingCallback secondCallbackThatWillBeRemoved = new TrackingCallback();
+                TrackingCallback thirdCallback = new TrackingCallback();
+                channel.addReturnListener(firstCallbackThatIsRegistedTwice);
+                channel.addReturnListener(firstCallbackThatIsRegistedTwice);
+                channel.addReturnListener(r -> {throw new RuntimeException("Listener throwing exception");});
```
Comment (ledoyen) [inline src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/ledoyen/rabbitmq-mock/pullRequest?prid=4930446)

**Comment by ledoyen** [comment]
Hi, sorry for the late response.

Thank you for this PR, code looks good !

Happy to merge !

**Comment by janssk1** [comment]
Thanks. Any plans for a new release?

**Comment by ledoyen** [comment]
Yes, **1.0.14** has been pushed to sonatype repository.

It should be available within 2 hours in Maven Central.

**Pull request merged**